test/test-network/systemd-networkd-tests.py

```diff
@@ -2311,5 +2343,26 @@ def test_dhcp_client_route_remove_on_renew(self):
         self.assertRegex(output, f'192.168.5.1 proto dhcp scope link src {address2} metric 1024')
 
 if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--build-dir', help='Path to build dir', dest='build_dir')
+    parser.add_argument('--networkd', help='Path to systemd-networkd', dest='networkd_bin', default=networkd_bin)
+    parser.add_argument('--wait-online', help='Path to systemd-networkd-wait-online', dest='wait_online_bin', default=wait_online_bin)
+    parser.add_argument('--networkctl', help='Path to networkctl', dest='networkctl_bin', default=networkctl_bin)
+    parser.add_argument('--valgrind', help='Enable valgrind', dest='use_valgrind', type=bool, nargs='?', const=True, default=use_valgrind)
+    parser.add_argument('--log-debug', help='Generate debugging logs', dest='enable_debug', type=bool, nargs='?', const=True, default=enable_debug)
+    ns, args = parser.parse_known_args(namespace=unittest)
+
+    if ns.build_dir:
```

Comment:
I think `--build-dir` is a useful shortcut but it seems to be confusing that it takes precedence over `--networkd`, `--wait-online` and `--networkctl`. It should probably at least somehow make it clear that these options can't be used together.

Reply:
A warning has been added.